core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -120,6 +123,16 @@
         this.useHttp1Pipelining = useHttp1Pipelining;
         this.idleTimeoutMillis = idleTimeoutMillis;
         this.pingIntervalMillis = pingIntervalMillis;
+
+        switch (proxyConfig.proxyType()) {
+            case SOCKS4:
+            case SOCKS5:
+            case CONNECT:
+                useProxyConnection = true;
+                break;
+            default:
+                useProxyConnection = false;
+        }
```

Comment:
nit: Not strong here 😉
```
useProxyConnection = proxyConfig.proxyType() != ProxyType.DIRECT;
```

Reply:
better using explicit `case` and throwing something at `default`?

Reply:
> useProxyConnection = proxyConfig.proxyType() != ProxyType.DIRECT;

A user might implement his own `ProxyConfig` and returns `null` for `proxyType()`. So we should check it explicitly.

> better using explicit case and throwing something at default?

We've already gone through the logic [here](
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java#L170) which just leaves a warning log.
I thought we don't have to leave a log twice. 😄